### DISCUSSION TITLE
Flashing issue with AI deck revision C1 and MacBook: freezing at 99% and exiting bootloader mode

### DISCUSSION THREAD

**Discussion by ArtemRedchych** [bitcraze/discussions#574] [Q&A]
I'm following the ["Getting started with the AI deck"](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/) tutorial, and I have a MacBook and an AI deck revision C1. I skipped the [step](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#gap8-bootloader-for-the-aideck-1-0-or-aideck-1-1-rev-a-c-) to flash the Gap8 bootloader separately, and now I'm having trouble[ flashing](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#flash-wifi-example) the "aideck_gap8_wifi_img_streamer_with_ap.bin" firmware onto the deck. The flashing process freezes at:

\ 95% Writing to bcAI:gap8 deck memory
| 96% Writing to bcAI:gap8 deck memory
/ 97% Writing to bcAI:gap8 deck memory
- 98% Writing to bcAI:gap8 deck memory
\ 99% Writing to bcAI:gap8 deck memory

, and the drone exits bootloader mode. Could this issue be related to the skipped step? And is it necessary to flash the Gap8 bootloader on an AI deck revision C1?

config:
Cfclient
Cfclient version: 2022.9
System: darwin
Python: 3.9.6
Qt: 5.15.2
PyQt: 5.15.7

Interface status
radio: Crazyradio version 0.53
UsbCdc: No information available
udp: None
prrt: No information available
cpx: None

Input readers
PySDL2 (0 devices connected)

Input devices
None

Crazyflie
Connected: radio://0/80/2M/E7E7E7E7E5
Firmware: 513c2815bf0a (CLEAN)

Decks found
bcAI: rev=C1, adr=0D5104BF00000035

Sensors found
BMP388: 1
imuPhi: 0.0
imuTheta: 0.0
imuPsi: 0.0
AK8963: 0
LPS25H: 0

Sensors tests
MPU6500: 0
AK8963: 0
LPS25H: 0
imuPhi: 0.0
imuTheta: 0.0
imuPsi: 0.0

**Comment by krichardsson** [comment]
> Could this issue be related to the skipped step? And is it necessary to flash the Gap8 bootloader on an AI deck revision C1?

Yes and yes :-)

**Reply by ArtemRedchych** [reply]
Can I use this programmer https://www.segger.com/products/debug-probes/j-link/models/j-link-edu-mini/ ?

**Reply by knmcguire** [reply]
not sure! but if it says it has full jlink functionality then it should? We haven't tried these ourselves

**Reply by mjm41** [reply]
Hi, I'm having the same issues. Freezing at 99% and I also skipped the programmer step. My AI Deck is also rev C1. Is there a more detailed explanation of the programming step? I bought the Olimex ARM-USB-TINY-H JTAG as suggested but didn't realize (first timer here) that you actually need a few more parts. Will that device work if I connect it to the [ARM-JTAG-20-10](https://www.olimex.com/Products/ARM/JTAG/ARM-JTAG-20-10/) connecting it with the AI Deck and the [CABLE-USB-A-B-1.8M](https://www.olimex.com/Products/Components/Cables/CABLE-USB-A-B-1.8M/) connecting the JTAG to my pc?

**Reply by ArtemRedchych** [reply]
Yes, it is exactly what i have used

**Reply by mjm41** [reply]
What did you use to connect the ARM-JTAG-20-10 with the Olimex ARM-USB-TINY-H JTAG? They both have female ends.
